src/GraphQL/Internal/Execution.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE PatternSynonyms #-}
+-- | Implement the \"Execution\" part of the GraphQL spec.
+--
+-- Actually, most of the execution work takes place in 'GraphQL.Resolver', but
+-- there's still a fair bit required to glue together the results of
+-- 'GraphQL.Internal.Validation' and the processing in 'GraphQL.Resolver'.
+-- This module provides that glue.
+module GraphQL.Internal.Execution
+  ( VariableValues
+  , ExecutionError(..)
+  , formatError
+  , getOperation
+  , substituteVariables
+  ) where
+
+import Protolude hiding (Type)
+
+import qualified Data.Map as Map
+import GraphQL.Value
+  ( Name
+  , Value
+  , pattern ValueNull
+  , Value'(..)
+  , List'(..)
+  , Object'(..)
+  )
+import GraphQL.Internal.Validation
+  ( Operation
+  , QueryDocument(..)
+  , VariableDefinition(..)
+  , VariableValue
+  , Variable
+  , Type(..)
+  )
+
+-- Maybe we should have?
+--
+-- >>> data Request = Request Schema Document (Maybe Operation) (Maybe VariableValues) InitialValue
+--
+-- And then we can have:
+--
+-- execute :: Request -> m Response
```

Comment:
That could work though the use would be a bit weird:

```haskell
let executor = execute @MiniAPI miniHandler
for incoming $ \request -> do
   executor request
```

That's why it's currently called `buildResover` so we can do `let resolve = buildResolver @MiniAPI miniHandler`.

But also not sure. We can try a few things to see what works.